src/test/groovy/example/http/HttpKit.java

```diff
@@ -0,0 +1,48 @@
+package example.http;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.Optional;
+
+public class HttpKit {
+    /**
+     * We should not have to write this method but to get the query string ONLY parameter
+     * in Http Servlet spec world means parsing the query string on POST
+     *
+     * @param request       the http servlet request
+     * @param parameterName the name of the parameter
+     *
+     * @return an optional parameter
+     */
+    public static Optional<String> getQueryParameter(HttpServletRequest request, String parameterName) {
+        if ("POST".equalsIgnoreCase(request.getMethod())) {
+            parameterName = parameterName + "=";
```

Comment:
we do but there is an edge case according to the spec.

if its POST BUT there is a query string parameter called query, you should take that instead

Its kinda confusing.  Perhaps we could drop this slavish following of the spec for example purposes